### PR TITLE
Eager load GET /comments ("All Comments")

### PR DESCRIPTION
**Pull request by TLSM**
Before vs after:

```
GET /comments
|----------|--------|--------|--------|--------|--------|------------|
| Database | SELECT | INSERT | UPDATE | DELETE | Totals | Duplicates |
|----------|--------|--------|--------|--------|--------|------------|
| default  |  167   |   0    |   0    |   0    |  167   |    154     |
|----------|--------|--------|--------|--------|--------|------------|

GET /comments
|----------|--------|--------|--------|--------|--------|------------|
| Database | SELECT | INSERT | UPDATE | DELETE | Totals | Duplicates |
|----------|--------|--------|--------|--------|--------|------------|
| default  |   13   |   0    |   0    |   0    |   13   |     0      |
|----------|--------|--------|--------|--------|--------|------------|
```

### CODE REVIEW

File: files/routes/front.py [hunk 1]
```diff
@@ -384,27 +388,31 @@ def random_user(v):
...
-	try: gt=int(request.values.get("after", 0))
-	except: gt=0
-
-	try: lt=int(request.values.get("before", 0))
-	except: lt=0
-
-	idlist = get_comments_idlist(v=v, page=page, sort=sort, t=t, gt=gt, lt=lt)
-	comments = get_comments(idlist, v=v)
-
+	page = request.values.get("page", 1, int)
```
Comment (justcool393) [inline files/routes/front.py]: this should still have a max in here

File: files/routes/front.py [hunk 2]
```diff
@@ -384,27 +388,31 @@ def random_user(v):
...
+	sort = request.values.get("sort", "new")
+	time_filter = request.values.get("t", defaulttimefilter)
+	time_gt = request.values.get("after", 0, int)
+	time_lt = request.values.get("before", 0, int)
```
Comment (justcool393) [inline files/routes/front.py:395]: `time_gt` and `time_lt` should honestly probably be clamped too tbh

**Review by justcool393 — CHANGES_REQUESTED** [review]
mostly good. there is a minor regression in that a 500 is possible though in this setup if you put in a page number that is a valid number but is less than 1

File: files/routes/front.py (hunk 1 above)
Comment (TLSM) [reply files/routes/front.py]: Done, the Postgres driver complained when it got a negative offset and crashed the request. Good find

File: files/routes/front.py (hunk 2 above)
Comment (TLSM) [reply files/routes/front.py:395]: Postgres behaves well for negative values and values > 2^32 for these. The actual column is a signed 32-bit int, though the query seems to not care about values outside those bounds. It's functionally useless but still well-formed. I tried all the way to ±(1 << 129) with no problems.

I guess there's a highly speculative DoS from parsing very large decimal strings into numeric representations, though if that's a risk, we've got a _lot_ of those floating around and I'd like a more general solution.

**Review by justcool393 — APPROVED** [review]
lg2m :)

**Review by zorbathut — APPROVED** [review]

**Pull request merged**